src/event.rs

```diff
@@ -118,64 +108,252 @@ pub enum Event<'a, T: 'static> {
-                event: event.clone(),
-            },
-            NewEvents(cause) => NewEvents(cause.clone()),
-            MainEventsCleared => MainEventsCleared,
-            RedrawRequested(wid) => RedrawRequested(*wid),
-            RedrawEventsCleared => RedrawEventsCleared,
-            LoopDestroyed => LoopDestroyed,
-            Suspended => Suspended,
-            Resumed => Resumed,
-        }
-    }
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum AppEvent {
+    /// Emitted when the application has been suspended.
+    Suspended,
+
+    /// Emitted when the application has been resumed.
+    Resumed,
+}
+
+/// Describes an event from a `Window`.
+#[derive(Debug, PartialEq)]
+pub enum WindowEvent<'a> {
+    /// The size of the window has changed. Contains the client area's new dimensions.
+    Resized(PhysicalSize<u32>),
+
+    /// The position of the window has changed. Contains the window's new position.
+    Moved(PhysicalPosition<i32>),
+
+    /// The window has been requested to close.
+    CloseRequested,
+
+    /// The window has been destroyed.
+    Destroyed,
+
+    /// A file is being hovered over the window.
+    ///
+    /// When the user hovers multiple files at once, this event will be emitted for each file
+    /// separately.
+    FileHovered(PathBuf),
+
+    /// A file has been dropped into the window.
+    ///
+    /// When the user drops multiple files at once, this event will be emitted for each file
+    /// separately.
+    FileDropped(PathBuf),
+
+    /// A file was hovered, but has exited the window.
+    ///
+    /// There will be a single `HoveredFileCancelled` event triggered even if multiple files were
+    /// hovered.
+    FileHoverCancelled,
+
+    /// The window gained focus.
+    FocusGained,
+
+    /// The window lost focus.
+    FocusLost,
+
+    Key(KeyEvent),
+
+    /// The window received a unicode character.
+    CharReceived(char),
+
+    /// The keyboard modifiers have changed.
+    ///
+    /// This is tracked internally to avoid tracking errors arising from modifier key state changes when events from
+    /// this device are not being delivered to the application, e.g. due to keyboard focus being elsewhere.
+    ///
+    /// Platform-specific behavior:
+    /// - **Web**: This API is currently unimplemented on the web. This isn't by design - it's an
+    ///   issue, and it should get fixed - but it's the current state of the API.
+    ModifiersChanged(ModifiersState),
+
+    PointerCreated(PointerId),
+    PointerEntered(PointerId),
+    PointerForce(PointerId, Force),
+    PointerTilt(PointerId, PointerTiltEvent),
+    PointerTwist(PointerId, f64),
+    PointerContactArea(PointerId, PhysicalSize<f64>),
+    PointerMoved(PointerId, PhysicalPosition<f64>),
+    // TODO: IMPLEMENT. IS THIS REASONABLE TO IMPLEMENT ON NON-WINDOWS PLATFORMS?
+    // PointerHovered(PointerId),
+    PointerButton(PointerId, PointerButtonEvent),
+    PointerDestroyed(PointerId),
+    PointerLeft(PointerId),
```

Comment:
These fields lack documentation. Do they work as follows?

- Create a pointer corresponding to the mouse on app start
- Create a pointer corresponding to a touch (with unique ID) when the touch starts and destroy when the touch ends
- Send `PointerButton` events just after creating and before destroying every touch event
- The same for a pen, if it exists (since in general one can't provide `PointerMoved` events for a pen not touching the screen)

For motion events, presumably the app must remember the last position for each pointer if it wishes to know relative motion? As you see [here](https://github.com/kas-gui/kas/blob/master/src/event/events.rs#L112) I ended up providing both absolute and relative positions for motion events in my unified API (but this API is for higher-level usage and misses some of the details).

How do the pen events (force, tilt, twist) work for sensitive pens? Do they tend to send all of these events each frame? If so it may be more convenient to receive all this data simultaneously. (But I'm only guessing since I haven't worked with pens.)